src/app/pages/plans/plan/plan.page.ts

```diff
@@ -11,25 +19,125 @@ import 'nve-designsystem/components/nve-checkbox/nve-checkbox.component.js';
+    },
+  ],
   styleUrl: './plan.page.css',
 })
 export class PlanPage {
-  geoJSON = inject(GeoJSONService);
+  geoJSONService = inject(GeoJSONService);
+  private mapLayersService = inject(MapLayersService);
   router = inject(Router);
   id = input.required<string>();
+  private map?: L.Map;
 
-  itemMetadata = computed(() => this.geoJSON.metadata().find((m) => m.id === this.id()));
+  private layerGroup = L.layerGroup();
+
+  itemMetadata = computed(() => this.geoJSONService.metadata().find((m) => m.id === this.id()));
 
   name = linkedSignal<string>(() => this.itemMetadata()?.name || '');
   comment = linkedSignal<string>(() => this.itemMetadata()?.comment || '');
   visibleOnMap = linkedSignal<boolean>(() => this.itemMetadata()?.visibleOnMap || false);
 
+  mapOptions = computed(() => {
+    const mapConfig = this.mapLayersService.mapConfig();
+    const layersConfig = mapConfig.map((map) => ({
+      layerId: map.layer,
+      layerConfig: settings.map.tiles.topoMapLayers[map.layer],
+    }));
+    const layers = layersConfig.map((x) =>
+      L.tileLayer(x.layerConfig.url, {
+        minZoom: settings.map.tiles.minZoom,
+        maxZoom: settings.map.tiles.maxZoom,
+        updateWhenIdle: settings.map.tiles.updateWhenIdle,
+        ...x.layerConfig.options,
+      })
+    );
+    const mapSettings: L.MapOptions = {
+      zoom: settings.map.tiles.zoomLevelObservationList,
+      maxZoom: settings.map.tiles.maxZoom,
+      minZoom: 8,
+      bounceAtZoomLimits: false,
+      attributionControl: false,
+      zoomControl: false,
+      layers,
+    };
+    return mapSettings;
+  });
+
+  async calculateAndCenterMap() {
+    const geoJSON = await this.geoJSONService.get(this.id());
+
+    if (!geoJSON || !this.map) return;
+
+    this.layerGroup.clearLayers();
+    const geoJsonLayer = L.geoJSON(geoJSON);
+    this.layerGroup.addLayer(geoJsonLayer);
+
+    if (!this.map.hasLayer(this.layerGroup)) {
+      this.layerGroup.addTo(this.map);
+    }
+
+    const lineFeatures = geoJSON.features.map((f) => {
+      if (f.geometry && f.geometry.type === 'LineString') {
+        return f.geometry as LineString;
+      }
+      return null;
+    });
+    const pointFeatures = geoJSON.features.map((f) => {
+      if (f.geometry && f.geometry.type === 'Point') {
+        return f.geometry as Point;
+      }
+      return null;
+    });
+
+    const lineCoords: [number, number][] = lineFeatures
+      .filter((f): f is LineString => f !== null)
+      .flatMap((f) => f.coordinates as [number, number][]);
+
+    const pointCoords: [number, number][] = pointFeatures
+      .filter((f): f is Point => f !== null)
+      .map((f) => f.coordinates as [number, number]);
+
+    const flatCoords: [number, number][] = [...lineCoords, ...pointCoords];
+
+    const latLngs = flatCoords.map(([lng, lat]) => L.latLng(lat, lng));
+
+    this.map.fitBounds(L.latLngBounds(latLngs), { padding: [5, 5] });
```

Comment:
Jeg tror dette kan forenkles til:

```suggestion
    this.map.fitBounds(geoJsonLayer.getBounds(), { padding: [5, 5] });
```

Eventuelt med sjekk på om bounds fra geoJsonLayer.getBounds() er valid (bounds.isValid())